jest.config.ts

```diff
@@ -16,6 +16,7 @@ const config: Config.InitialOptions = {
   transform: {
     '\\.hbs$': 'jest-text-transformer',
   },
+  openHandlesTimeout: 5000,
```

Comment:
There is no problem. Jest's default `openHandlesTimeout` is one second and it isn't enough for all the connections opened by `graphql-sse` to close. It's a `graphql-sse` library thing. If you want, you can run our test suite with the [--detectOpenHandles](https://jestjs.io/docs/cli#--detectopenhandles) flag, and you'll see that it actually doesn't print out any open handles and finishes nicely. The problem with using that flag as the solution is that it requires `--runInBand` so tests _must_ run in series, and I'm [currently working on parallelizing tests](https://github.com/FuelLabs/fuels-ts/tree/ns/feat/parallelize-tests).

The truest alternative to this would be us implementing our own bespoke library for handling subscriptions in order to have full ownership of the connection and close it when we wish. IMO it's totally not necessary as `graphql-sse` does its job well, just with this connection closing delay quirk attached. 

Or we could implement some "open connection tracker" which would wait for all the connections opened by `graphql-sse` to close before `jest` ends and then be sure that this `graphl-sse`-specific quirk is monitored; but again, I don't see the point in spending energy on that when we can just tell jest to wait four more seconds than the default and the connections would close on their own. As [this SO comment](https://stackoverflow.com/a/75949006) says, it's not unheard of that jest's defaults sometimes don't cut it (e.g. a TCP socket can take longer to close than jest expects). 

In the end, the most important thing from a user and fuel node's standpoint is that the connections _do_ get closed so there is no _real_ problem. This is just a change in `jest`'s config to stop it from giving us false negatives that we know are false.

Reply:
Or we need a method that closes all active connections that we can call from a  Jest's teardown hook.

@camsjams Makes sense?

Reply:
That would defeat the purpose of jest detecting open handles. The 5s threshold allows `graphql-sse` to close its connections, but we could still introduce some change into our code that keeps connections open even longer - something we'd be like to be notified of. 
If we'd go down the route of closing all connections from a teardown hook, then we could just not do that and ignore this warning altogether by setting `openHandlesTimeout` to zero (see [docs](https://jestjs.io/docs/configuration#openhandlestimeout-number))?

Reply:
> Well adding this timeout of adds 5s to all tests forever since this issue is not solved.

@camsjams It adds five seconds at the end after all tests are run, it's not a per-test config. Five seconds isn't intolerable when the test suite takes 100+ seconds.

> I'd prefer to dig deeper here since the handles not being closed is the issue, not Jest. This means there are hanging connections that could affect real-world applications.

I already dug deep. The handles _do_ get closed, just not in the time that Jest expects them to. No warning after 5s is a testament to that. 

You can look at the `TransactionResponse` test I added. If you remove the last `cleanup()` call in it, jest _will not exit at all_ because of the open handle created by `launchNode`. However, if you leave the test as it is (with `cleanup` called) but remain with the default one second for `openHandlesTimeout` in `jest.config.ts`, jest will warn of open handles (because of the subscriptions), but it _will exit_. This means that the open handles got closed in the meantime and jest could exit. It just didn't exit in the time it expected _by default_ and thus warned of a _possible_ issue. 

Another way to know that the handles indeed do get closed is by modifying the test and adding this code before the `cleanup()` call:

```ts
    const subscription = nodeProvider.operations.statusChange({ transactionId });

    /**
     * graphql-sse returns AsyncIterableIterator which has .next(), .return() and .error() on it.
     * .next() is called implicitly in a for-await-of loop when iterating over the iterator.
     * .return() is implicitly called when a `break` is hit in the for-await-of loop.
     * The .return() call in graphql-sse disposes of the subscription.
     */

    // @ts-expect-error .next() exists in runtime, as explained above
    await subscription.next();

    // @ts-expect-error .return() exists in runtime, as explained above
    await subscription.return();

    cleanup();
```
These methods are in essence what gets called in `TransactionResponse.waitForResult`, just implicitly in the `for-await-of` loop.

This code will finish successfully and jest will exit. However, if you remove the call to `await subscription.return()`, the test will finish successfully, but jest won't exit. This is because there is still an open connection which would've been closed by the call  `await subscription.return()`.

Reply:
Okay, so it seems that we have misunderstood the `openHandlesTimeout` flag related to test time. On this branch, I have just set its value from 5s to 100s and ran `pnpm jest --no-cache --runInBand packages/fuel-gauge/src/vector-types.test.ts` and the test time wasn't affected _at all_. What this flag does is it tells jest to wait the assigned time before warning of open handles. If the handles close in the meantime, the tests will end - unaffected by the flag. The tests will end in the same time both when the `openHandlesTimeout` flag is 0.001 second and 100 seconds; the only difference between those two runs would be _when_ jest warns of open handles.

So with this newfound knowledge and the fact that the library itself does close the handles in the end, are we on the same page?